cosmos/providers/dbt/core/operators.py

```diff
@@ -179,7 +191,13 @@ def add_global_flags(self):
         flags = []
         for global_flag in global_flags:
             dbt_name = f"--{global_flag.replace('_', '-')}"
-            global_flag_value = self.__getattribute__(global_flag)
+            # intercept project directory and route it to r/w tmp
+            if global_flag == "project_dir":
+                global_flag_value = (
+                    f"/tmp/dbt/{os.path.basename(self.__getattribute__(global_flag))}"
```

Comment:
I like the idea of using the existing path name in this path. Leaves the option for multiple simultaneous projects. Is there any risk of cross contamination if there are multiple users or multiple airflows? Maybe we make the base path configurable?

Reply:
I don't know if I'm following. Let's put it into context of the jaffle_shop project to ensure we are on the same page. So for this example the project_dir value is `jaffle_shop`, so the `jaffle_shop` directory from `./dags/dbt/jaffle_shop` will get written to `/tmp/dbt/jaffle_shop`. There are file lock mechanisms to ensure that multiple workers don't colide when that copy happens from `/dags` to `/tmp`. I've confirmed this by deleting `/tmp` and executing several `DbtRunOperators` in parallel to ensure they don't get hung-up on each other. I think this will remain true regardless of how many users or Airflows there are running this against tmp. Does that answer your question?

Reply:
@dimberman forgot to @ you, sorry!